autoload/beancount.vim

```diff
@@ -56,14 +66,29 @@ function! beancount#complete_account(findstart, base)
         endif
         let b:beancount_accounts = beancount#find_accounts(l:root)
     endif
-    let l:pattern = '\V' . substitute(a:base, ":", '\\S\\*:\\S\\*', "g")
+
+    if g:beancount_account_completion ==? 'chunks'
+        let l:pattern = '\V' . substitute(a:base, ":", '\\[^ \\t:]\\*:', "g") . '\[^ \t:]\*'
+    else
+        let l:pattern = '\V' . substitute(a:base, ":", '\\S\\*:\\S\\*', "g")
+    endif
+
     let l:matches = []
     let l:index = -1
     while 1
         let l:index = match(b:beancount_accounts, l:pattern, l:index + 1)
         if l:index == -1 | break | endif
-        call add(l:matches, b:beancount_accounts[l:index])
+        if g:beancount_account_completion ==? 'chunks'
+            call add(l:matches, matchstr(b:beancount_accounts[l:index], l:pattern))
```

Comment:
By appending `\.\*` to the regular expression in the default case, you can use this branch every time and avoid the extra if statement.